src/Packagist/WebBundle/Controller/PackageController.php

```diff
@@ -1120,6 +1138,45 @@ public function versionStatsAction(Request $req, Package $package, $version)
         return $this->overallStatsAction($req, $package, $version);
     }
 
+    /**
+     * @Route(
+     *      "/packages/{name}/advisories",
+     *      name="view_package_advisories",
+     *      requirements={"name"="([A-Za-z0-9_.-]+/[A-Za-z0-9_.-]+?|ext-[A-Za-z0-9_.-]+?)"}
+     * )
+     */
+    public function securityAdvisoriesAction(Request $request, $name)
+    {
+        /** @var SecurityAdvisoryRepository $repo */
+        $repo = $this->getDoctrine()->getRepository(SecurityAdvisory::class);
+        $securityAdvisories = $repo->getPackageSecurityAdvisories($name);
+        $advisoryCount = count($securityAdvisories);
+
+        $paginator = new Pagerfanta(new FixedAdapter($advisoryCount, $securityAdvisories));
+        $data['securityAdvisories'] = $paginator;
+        $data['count'] = $advisoryCount;
+        $data['name'] = $name;
+
+        $data['matchingAdvisories'] = [];
+        if ($versionId = $request->query->getInt('version')) {
+            $version = $this->getDoctrine()->getRepository(Version::class)->findOneBy([
+                'name' => $name,
+                'id' => $versionId,
+            ]);
+            if ($version) {
+                foreach ($securityAdvisories as $advisory) {
+                    $versionParser = new VersionParser();
+                    $affectedVersionConstraint = $versionParser->parseConstraints($advisory['affectedVersions']);
+                    if (!isset($data['hasVersionSecurityAdvisories'][$version->getId()]) && $affectedVersionConstraint->matches(new Constraint('=', $version->getNormalizedVersion()))) {
+                        $data['matchingAdvisories'] = $advisory['id'];
+                    }
+                }
+            }
+        }
```

Comment:
Generally speaking once this block executes, the paginator will be in a bad state I assume, as this filters the list of securityAdvisories which are displayed? Might have missed something, but it sounds to me like the filtering should happen before paginator and count etc are assigned to $data?

Reply:
The pagination is currently unused. Removed it completely.